setup.py

```diff
@@ -14,6 +14,8 @@
 
 os.chdir(os.path.split(os.path.abspath(__file__))[0])
 
+# Try to get and update the version.
+# N.B. Similar code appears in docs/conf.py.
```

Comment:
Or even just a top-level `version_gen.py`, since it would only be used by `setup.py` and `conf.py` anyway.